src/main/kotlin/com/zaxxer/ping/impl/util/WaitingTargetCollection.kt

```diff
@@ -50,12 +52,9 @@ class WaitingTargetCollection {
     fun remove(sequence: Short): PingTarget? {
         val pingTarget = waitingTargetMap.remove(sequence) ?: return null
         pingTarget.complete = true
+        targetTimeoutQueue.remove(pingTarget)
```

Comment:
The call here to ``targetTimeoutQueue.remove(pingTarget)`` is unnecessary, as the item will be removed from ``targetTimeoutQueue`` due to ``pingTarget.complete=true`` in ``peekTimeoutQueue()`` above. Basically, this marks it complete and it will be "cleaned up" by timeout processing.